File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
//品牌控制层
app.controller('brandController',function ($scope,$controller,$http,brandService) {
    $controller('baseController',{$scope:$scope});//继承baseController控制器；伪继承；就是传递@scope
    // //查询品牌列表
    // $scope.findAll=function(){
    // 	$http.get('../brand/findAll.do').success(
    // 			function(data){
    // 				$scope.list=data;
    // 			});
    // }



    // $scope.findPage=function(pageNum,pageSize){
    // 	$http.get('../brand/findPage.do?pageNum='+pageNum+'&pageSize='+pageSize).success(
    // 			function(data){
    // 				$scope.list=data.rows;//当前页的数据
    // 				$scope.paginationConf.totalItems=data.total;
    // 			});
    // };

    //增加品牌
    $scope.save=function(){
        var object=null;
        if($scope.entity.id!=null){
            object=brandService.update($scope.entity);
        }else{
            object=brandService.add($scope.entity);
        }
        object.success(
            function (data) {
                if (data.success){
                    $scope.reloadList();//重新加载
                    alert(data.message)
                }else{
                    alert(data.message)
                }
            });
    };
    //查找数据回显示--跟新
    $scope.findOne=function(id){
        brandService.findOne(id).success(
            function(data){
                $scope.entity=data;
            }
        );
    };
    //向后台发送删除请求
    //删除
    $scope.deleteBrands=function(){
        if(confirm('确定要删除吗？')){
            brandService.deleteBrands($scope.selectIds).success(
                function(response){
                    if(response.success){
                        $scope.reloadList();//刷新
                    }else{
                        alert(response.message);
                    }
                }
            );
        }

    };

    //查询
    $scope.searchEntity={};
    $scope.search=function (pageNum,pageSize) {
        brandService.search(pageNum,pageSize,$scope.searchEntity).success(
            function(data){
                $scope.list=data.rows;//当前页的数据
                $scope.paginationConf.totalItems=data.total;
            });
    }
});

File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;

import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import entity.PageResult;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;



@RestController
@RequestMapping("/brand")
public class    BrandController {

    @Reference(timeout = 10000)
    private BrandService brandService;

//    @RequestMapping("/findAll")//查询所有品牌
//    public List<TbBrand> findAll(){
//        return brandService.findAll();
//    }

//    @RequestMapping("/findPage")
//    public PageResult findPage(int pageNum,int pageSize) {
//        return brandService.findPage(pageNum,pageSize);
//
//    }
    @RequestMapping("/add")
    public Result addBrand(@RequestBody  TbBrand brand){
        try {
            brandService.addBrand(brand);
            return  new Result(true,"添加成功");
        }catch (Exception e){
            e.printStackTrace();
            return  new Result(true,"添加成功");
        }
    }
    @RequestMapping("/findOne")
    public TbBrand findOneById(long id) {
      return  brandService.findOneById(id);
    }


    @RequestMapping("/update")
    public Result updateBrand(@RequestBody  TbBrand brand){
        try {
            brandService.updateBrand(brand);
            return  new Result(true,"修改成功");
        }catch (Exception e){
            e.printStackTrace();
            return  new Result(true,"修改成功");
        }
    }

    @RequestMapping("/delete")
    public Result delteBrand(Long[] ids) {
        try {
            brandService.delteBrand(ids);
            return  new Result(true,"删除成功");
        }catch (Exception e){
            e.printStackTrace();
            return  new Result(true,"删除成功");
    }
}
    @RequestMapping("/search")
    public PageResult search(@RequestBody  TbBrand brand,int pageNum,int pageSize){
        return  brandService.findPage(brand,pageNum,pageSize);
    }

    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList() {
        return brandService.selectOptionList();
    }
}
